manim/utils/space_ops.py

```diff
@@ -581,49 +530,47 @@ def line_intersection(line1: Sequence[float], line2: Sequence[float]) -> np.ndar
     ------
     ValueError
         Error is produced if the two lines don't intersect with each other
+        or if the coordinates don't lie on the xy-plane.
     """
-    x_diff = (line1[0][0] - line1[1][0], line2[0][0] - line2[1][0])
-    y_diff = (line1[0][1] - line1[1][1], line2[0][1] - line2[1][1])
+    if any(np.array([line1, line2])[:, :, 2].reshape(-1)):
+        # checks for z coordinates != 0
+        raise ValueError("Coords must be in the xy-plane.")
+
+    # algorithm from https://stackoverflow.com/a/42727584
+    padded = (
+        np.pad(np.array(i)[:, :2], ((0, 0), (0, 1)), constant_values=1)
+        for i in (line1, line2)
+    )
+    line1, line2 = (np.cross(*i) for i in padded)
+    x, y, z = np.cross(line1, line2)
 
-    def det(a, b):
-        return a[0] * b[1] - a[1] * b[0]
+    if z == 0:
+        raise ValueError("Lines do not intersect.")
```

Comment:
```suggestion
        raise ValueError("Line segments do not intersect.")
```

Reply:
The rest seem appropriate, but this one implies that the segments themselves not intersecting would raise this error which isn't true. How about "The line segments are parallel." since it's the only time the error appears?

Reply:
I thought from your previous comment, the function does not find a result if the intersection point is not within the line segments?

Reply:
> The rest seem appropriate, but this one implies that the segments themselves not intersecting would raise this error which isn't true. How about "The line segments are parallel." since it's the only time the error appears?

Even if they are coincident not just parallel

Reply:
> I thought from your previous comment, the function does not find a result if the intersection point is not within the line segments

I must've said something wrong, my bad. Currently, the function acts as if the two points given for each 'line' are points along an infinite line. Therefore, only if the lines are parallel (or coincident), the error is raised.

The complete message could be: "No intersections found. The lines are parallel or coincident."

Reply:
Sounds good, except mentioning coincidence is redundant since coincident lines are just a special case of parallel lines.